Packs/Illumio/Integrations/IllumioCore/IllumioCore.py

```diff
@@ -22,6 +28,171 @@ def __init__(self, port):
         super().__init__(self.message)
 
 
+class InvalidSingleSelectException(Exception):
+    """Exception class for invalid single-select option."""
+
+    def __init__(self, value, argument, argument_list):
+        self.message = "{} is an invalid value for {}. Possible values are: {}.".format(value, argument, argument_list)
+        super().__init__(self.message)
+
+
+class InvalidMultiSelectException(Exception):
+    """Exception class for invalid multi-select option."""
+
+    def __init__(self, argument, argument_list):
+        self.message = "Invalid value for {}. Possible comma separated values are {}.".format(argument, argument_list)
+        super().__init__(self.message)
+
+
+''' HELPER FUNCTIONS '''
+
+
+def validate_arguments_for_traffic_analysis(port: Optional[int], policy_decisions: list, protocol: str):
+    """Validate arguments for illumio-traffic-analysis command.
+
+    Args:
+        port: Port number.
+        policy_decisions: Policy decision to include in the search result.
+        protocol: Communication protocol.
+
+    Returns:
+        str: Error if value of the argument is not valid.
+    """
+    if port < MIN_PORT or port > MAX_PORT:  # type: ignore
+        raise InvalidPortException(port)
+
+    for decision in policy_decisions:
+        if decision not in VALID_POLICY_DECISIONS:
+            raise InvalidMultiSelectException('policy_decisions', VALID_POLICY_DECISIONS)
+
+    if protocol not in VALID_PROTOCOLS:
+        raise InvalidSingleSelectException(protocol, "protocol", VALID_PROTOCOLS)
+
+
+def validate_arguments_for_illumio_virtual_service_create_command(name: str | None, port: str | None,
+                                                                  protocol: str) -> tuple:
+    """Validate and convert arguments for illumio virtual service create command.
+
+    Args:
+        name: Name of virtual service.
+        port: Port number.
+        protocol: Protocol name.
+
+    Returns:
+        Returns tuple of converted values (port, protocol).
+    """
+    validate_required_parameters(name=name, port=port)
+
+    port = arg_to_number(port, arg_name="port")  # type: ignore
+
+    if port != -1 and (port > MAX_PORT or port < MIN_PORT):  # type: ignore
+        raise ValueError("{} is an invalid value for port. Value must be in 1 to 65535 or -1.".format(port))
+
+    if protocol not in VALID_PROTOCOLS:
+        raise InvalidSingleSelectException(protocol, 'protocol', VALID_PROTOCOLS)
+
+    proto = convert_protocol(protocol)
+    return port, proto
+
+
+
+def prepare_hr_for_traffic_analysis(response: list, protocol: str) -> str:
```

Comment:
This type of function name `do_x_for` is too verbose, and specifically `prepare_hr_for` doesn't make much sense as it's not clear that `hr` is short for "human-readable output" until you read the function description. `prepare_traffic_analysis_output` or similar is better.

Reply:
I'll change it as per discussed above